tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -881,6 +886,172 @@ def _wait_for_traffic(self, producer, acked, timeout=1800):
                 f"Low throughput while preparing for the test: {_throughput} MB/s"
             ) from None
 
+    # returns a tuple of (replicas, readyReplicas)
+    def get_cluster_replicas(self, cluster_name: str) -> tuple[int, int]:
+        if self.operator_version == 2:
+            return self._get_cluster_replicas_v2(
+            ), self._get_cluster_ready_replicas_v2()
+        return self._get_cluster_replicas(
+            cluster_name), self._get_cluster_ready_replicas(cluster_name)
+
+    # sets new target replica count for the cluster
+    def set_cluster_replicas(self, cluster_name: str, new_replicas: int):
+        # for operator V2, we scale the statefulset because that seems
+        # to be the only way that works
+        if self.operator_version == 2:
+            self.redpanda.kubectl.cmd([
+                'scale', '-n=redpanda', 'statefulset', 'redpanda-broker',
+                f'--replicas={new_replicas}'
+            ])
+            return
+
+        # for operator v1, we patch the cluster CRD instead
+        patch = [{
+            'op': 'replace',
+            'path': '/spec/replicas',
+            'value': new_replicas
+        }]
+        patch_str = json.dumps(patch)
+
+        # kubectl patch -n=redpanda cluster rp-clkd0n22nfn1jf7vd9t0 --type=json -p='[{"op":"replace","path":"/spec/replicas","value":4}]'
+        self.redpanda.kubectl.cmd([
+            'patch', '-n=redpanda', 'cluster', cluster_name, '--type=json',
+            f"-p='{patch_str}'"
+        ])
+
+    def _kube_autoretry_int(self,
+                            command: list[str],
+                            max_retries: int = 10) -> int:
+        result = ''
+        retry_count = 0
+        while retry_count < max_retries:
+            result = self.redpanda.kubectl.cmd(command)
+            if type(result) is str and result != '':
+                return int(result)
+
+            retry_count += 1
+            self.logger.debug(
+                f'retrying `kubectl {command}` attempt #{retry_count}')
+            time.sleep(0.50)
+        raise TimeoutError(
+            f'hit max retries ({max_retries}) for `kubectl {command}`')
+
+    # waits until the Redpanda broker replica count reaches the target value
+    def wait_cluster_replicas(self, cluster_name: str, replicas: int):
+        if self.operator_version == 2:
```

Comment:
this overall pattern of having an `if` in each relevant method is pretty awkward. It also leads to weirdness where we have to pass in arguments like `cluster_name` which are ignored in the v2 context, which seems wrong: if the user is passing in a specific arguments there, they would expect it to be respected?

IMO it is better to move this functionality either into the kubectl tool, and have two flavors of that tool (sub classes) for V1 and V2 with the parts that are different in the subclass, or have another tool with two implementations (probably sharing a base ABC) which has kubectl as a member.

Reply:
This is especially important since this HTT class test fixture isn't the only test fixture that runs against the cloud: we will likely need these methods for other cloud-using tests like OMBValidationTest.

Reply:
I should have timeboxed this to a day, but I didn't, and the changes started to get a bit out of hand… so I created https://redpandadata.atlassian.net/browse/DEVPROD-2407 for us to tackle both HTT and OMB and their operator/cluster interactions.